controllers/clustercache/metrics.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clustercache
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	ctrlmetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+func init() {
+	// Register the metrics at the controller-runtime metrics registry.
+	ctrlmetrics.Registry.MustRegister(healthCheck)
+}
+
+var (
+	healthCheck = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "capi_clustercache_health_check",
+			Help: "Result of a single clustercache healthcheck.",
+		}, []string{
+			"cluster",
```

Comment:
Do you know if there's some sort of standard/convention on how to encode namespace + name into metric labels?

If there is, would be nice to follow it to make it easier to correlate / join metrics

Reply:
seems that `namespace` and then the object kind are common label keys. here's one specific example:
https://github.com/kubernetes/kubernetes/blob/5e0931336b54a9a0c38c37c7c1afe1ca41033c5f/pkg/kubelet/metrics/collectors/resource_metrics.go#L72-L77

in terms of CAPI setup, this might be a bit confusing since we are transforming pod metadata `namespace` as `namespace` from the metrics source:
https://github.com/kubernetes-sigs/cluster-api/blob/main/hack/observability/prometheus/values.yaml#L50-L52

and so in this case we would be sort of overloading the namespace key

maybe we should change that?

anyway,  pushed up a change to address your most recent comments and at least start splitting name/namespace for the clusters rather than just using the `.String()` method

Reply:
> maybe we should change that?

Maybe, not sure. This matches the Prometheus example config: https://github.com/prometheus/prometheus/blob/main/documentation/examples/prometheus-kubernetes.yml#L294-L297

Reply:
@cahillsf What do you think about cluster_name & cluster_namespace?

I think that makes both label keys a bit more consistent with each other

Reply:
sure sure yeah that sounds good